cdap-data-fabric/src/main/java/co/cask/cdap/data/format/RecordFormats.java

```diff
@@ -26,29 +30,56 @@
+   * format will be created, with specification settings applied on top of default settings.
+   * For example, "CSV" will map to the {@link DelimitedStringsRecordFormat}, with a comma as the delimiter,
+   * whereas "TSV" will map to the {@link DelimitedStringsRecordFormat}, with a tab as the delimiter.
+   * If the name is not a standard name, it is interpreted as a class name.
    *
-   * @param name Name of the format to instantiate.
-   * @param <FROM> Type of underlying object the format reads.
-   * @param <TO> Type of object the format reads the underlying object into.
-   * @return Instantiated {@link RecordFormat} based on the given name.
-   * @throws IllegalAccessException if there was an illegal access when instantiating the record format.
-   * @throws InstantiationException if there was an exception instantiating the record format.
-   * @throws ClassNotFoundException if the record format class could not be found.
+   * @param spec the specification for the format to create and initialize
+   * @param <FROM> Type of underlying object the format reads
+   * @param <TO> Type of object the format reads the underlying object into
+   * @return Initialized {@link RecordFormat} based on the given name
+   * @throws IllegalAccessException if there was an illegal access when instantiating the record format
+   * @throws InstantiationException if there was an exception instantiating the record format
+   * @throws ClassNotFoundException if the record format class could not be found
+   * @throws UnsupportedTypeException if the specification is not supported by the format
    */
-  public static <FROM, TO> RecordFormat<FROM, TO> create(String name)
-    throws IllegalAccessException, InstantiationException, ClassNotFoundException {
-    Class<? extends RecordFormat> formatClass = NAME_MAP.get(name);
-    return (RecordFormat<FROM, TO>) (formatClass == null ?
+  public static <FROM, TO> RecordFormat<FROM, TO> createInitializedFormat(FormatSpecification spec)
+    throws IllegalAccessException, InstantiationException, ClassNotFoundException, UnsupportedTypeException {
+    String name = spec.getName();
+
+    // check if it's a standard class
+    Class<? extends RecordFormat> formatClass = NAME_CLASS_MAP.get(name.toLowerCase());
+    RecordFormat<FROM, TO> format = (RecordFormat<FROM, TO>) (formatClass == null ?
       Class.forName(name).newInstance() : formatClass.newInstance());
+
+    // check if there should be some default settings
+    Map<String, String> defaultSettings = NAME_SETTINGS_MAP.get(name.toLowerCase());
+    if (defaultSettings != null) {
+      Map<String, String> mergedSettings = Maps.newHashMap(defaultSettings);
+      if (spec.getSettings() != null) {
+        mergedSettings.putAll(spec.getSettings());
+      }
+      FormatSpecification mergedSpec = new FormatSpecification(name, spec.getSchema(), mergedSettings);
```

Comment:
Don't you need to initialize the format with the mergedSpec here, or am I missing something?